Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2300,6 +2300,14 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
             }
         }
 
+        if (rightConst && rightConst.isNumber()) {
```

Comment:
We need to ensure that leftConst is some kind of number abstract value since if we have,
```
object < NaN
```
then, object.valueOf needs to be called. (side effectful).
Can you add a test for that?